src/main/java/it/gov/pagopa/rtd/ms/rtdmsfilereporter/domain/model/FileMetadata.java

```diff
@@ -35,4 +35,16 @@ public static FileMetadata createNewFileMetadata(String name) {
   public static FileMetadata createNewFileMetadataWithStatus(String name, FileStatusEnum status) {
     return new FileMetadata(name, null, status, LocalDateTime.now());
   }
+
+  /**
+   * Descending ordering for transmissionDate. The files will be ordered from the most recent to the
+   * oldest.
+   *
+   * @param o the object to be compared.
+   * @return result of compare.
+   */
+  @Override
+  public int compareTo(@org.jetbrains.annotations.NotNull FileMetadata o) {
```

Comment:
@and-mora can we use javax.validation.constraints.NotNull?

Reply:
good point. It has been autocompleted by the ide itself. I'm going to correct it.